.github/workflows/pr-scan.yml

```diff
@@ -0,0 +1,87 @@
+name: PR Security Scan
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+  workflow_dispatch:
+jobs:
+  scan:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+        submodules: 'recursive'
+    - name: Build the Docker image
+      uses: docker/build-push-action@v5
+      with: 
+        context: .
+        push: false
+        load: true
+        tags: pr_image:${{ github.sha }}
+    - name: Run Trivy vulnerability scanner on image
+      uses: aquasecurity/trivy-action@master
+      with:
+        image-ref: 'pr_image:${{ github.sha }}'
+        format: 'json'
+        output: 'trivy-image-results.json'
+        severity: 'CRITICAL,HIGH'
+    - name: Run Trivy vulnerability scanner on source code
```

Comment:
Were you able to look into what the result of generating an [SBOM](https://github.com/aquasecurity/trivy-action?tab=readme-ov-file#using-trivy-to-generate-sbom) was and if it was useful?

Reply:
Since this is PR-triggered, this would make sense in a nightly run